failing tests on raspberry pi

```
============================================= FAILURES ==============================================
_______________________________________ test_make_serial_spi ________________________________________

    def test_make_serial_spi():
        """
        :py:func:`luma.core.cmdline.make_serial.spi` returns an SPI instance.
        """
        factory = cmdline.make_serial(test_spi_opts)
        with pytest.raises(error.UnsupportedPlatform):
>           factory.spi()
E           Failed: DID NOT RAISE <class 'luma.core.error.UnsupportedPlatform'>

tests/test_cmdline.py:115: Failed
___________________________________ test_make_serial_spi_alt_gpio ___________________________________

    def test_make_serial_spi_alt_gpio():
        """
        :py:func:`luma.core.cmdline.make_serial.spi` returns an SPI instance
        when using an alternative GPIO implementation.
        """
        class opts(test_spi_opts):
            gpio = 'fake_gpio'
    
        with patch.dict('sys.modules', **{
                'fake_gpio': Mock(unsafe=True)
            }):
            factory = cmdline.make_serial(opts)
            with pytest.raises(error.DeviceNotFoundError):
>               factory.spi()
E               Failed: DID NOT RAISE <class 'luma.core.error.DeviceNotFoundError'>

tests/test_cmdline.py:131: Failed
______________________________________ test_create_device_oled ______________________________________

    def test_create_device_oled():
        """
        :py:func:`luma.core.cmdline.create_device` supports OLED displays.
        """
        display_name = 'oled1234'
        display_types = {'oled': [display_name]}
    
        class args(test_spi_opts):
            display = display_name
    
        module_mock = Mock()
        with patch.dict('sys.modules', **{
                'luma': module_mock,
                'luma.oled': module_mock,
                'luma.oled.device': module_mock
            }):
            with pytest.raises(error.UnsupportedPlatform):
>               cmdline.create_device(args, display_types=display_types)
E               Failed: DID NOT RAISE <class 'luma.core.error.UnsupportedPlatform'>

tests/test_cmdline.py:161: Failed
====================================== pytest-warning summary =======================================
WW0 in tests/test_cmdline.py:108 the following warning was recorded:
 luma/core/interface/serial.py:171: RuntimeWarning: This channel is already in use, continuing anyway.  Use GPIO.setwarnings(False) to disable warnings.
  self._gpio.setup(pin, self._gpio.OUT)

WW0 in tests/test_cmdline.py:189 the following warning was recorded:
 luma/core/serial.py:18: DeprecationWarning: luma.core.serial is deprecated and will be removed in v1.0.0: use luma.core.interface.serial instead
  'use luma.core.interface.serial instead')

====================== 3 failed, 114 passed, 2 pytest-warnings in 9.78 seconds ======================
```

Suspect that all three of these are using real spi or i2c (which is present on the pi, obvs), maybe they need to do something similar to `test_make_serial_spi_alt_gpio` that slots in a fake spi and i2c as well as the fake gpio.

TBH I'm having tremendous problems running the test on ax x86 linux machine at the moment:

```
=========================================================================== FAILURES ============================================================================
____________________________________________________________________ test_module_deprecated _____________________________________________________________________

    def test_module_deprecated():
        """
        The :py:mod:`luma.core.serial` module is deprecated.
        """
        msg = ("luma.core.serial is deprecated and will be removed in " +
            "v1.0.0: use luma.core.interface.serial instead")
    
>       with pytest.deprecated_call() as c:
E       TypeError: deprecated_call() takes at least 1 argument (0 given)

tests/test_serial.py:20: TypeError
__________________________________________________________________ test_init_params_deprecated __________________________________________________________________

    def test_init_params_deprecated():
        port = 5
        device = 2
        bus_speed = 16000000
        dc = 80
        rst = 90
        msg1 = 'bcm_DC argument is deprecated in favor of gpio_DC and will be removed in 1.0.0'
        msg2 = 'bcm_RST argument is deprecated in favor of gpio_RST and will be removed in 1.0.0'
    
>       with pytest.deprecated_call() as c:
E       TypeError: deprecated_call() takes at least 1 argument (0 given)

tests/test_spi.py:72: TypeError
_______________________________________________________________________ test_deprecation ________________________________________________________________________

    def test_deprecation():
        """
        ``deprecated`` creates a DeprecationWarning.
        """
        class DeprecatedClass(object):
            def __init__(self):
                self.msg = 'Deprecated; will be removed in 0.0.1'
                util.deprecation(self.msg)
    
>       with pytest.deprecated_call() as c:
E       TypeError: deprecated_call() takes at least 1 argument (0 given)

tests/test_util.py:30: TypeError
============================================================= 3 failed, 114 passed in 4.07 seconds ==============================================================
```

--- a/.codacy.yaml
+++ b/.codacy.yaml
@@ -1,0 +1,4 @@
+exclude_paths:
+  - tests/**
+  - .tox/**/*
+  - doc/**

--- a/tests/test_cmdline.py
+++ b/tests/test_cmdline.py
@@ -110,9 +110,12 @@ def test_make_serial_spi():
     """
     :py:func:`luma.core.cmdline.make_serial.spi` returns an SPI instance.
     """
-    factory = cmdline.make_serial(test_spi_opts)
-    with pytest.raises(error.UnsupportedPlatform):
-        factory.spi()
+    try:
+        factory = cmdline.make_serial(test_spi_opts)
+        assert 'luma.core.interface.serial.spi' in repr(factory.spi())
+    except error.UnsupportedPlatform as e:
+        # non-rpi platform, e.g. ubuntu 64-bit
+        pytest.skip(str(e))
 
 
 def test_make_serial_spi_alt_gpio():
@@ -126,9 +129,12 @@ def test_make_serial_spi_alt_gpio():
     with patch.dict('sys.modules', **{
             'fake_gpio': Mock(unsafe=True)
         }):
-        factory = cmdline.make_serial(opts)
-        with pytest.raises(error.DeviceNotFoundError):
-            factory.spi()
+        try:
+            factory = cmdline.make_serial(opts)
+            assert 'luma.core.interface.serial.spi' in repr(factory.spi())
+        except error.DeviceNotFoundError as e:
+            # non-rpi platform, e.g. ubuntu 64-bit
+            pytest.skip(str(e))
 
 
 def test_create_device():
@@ -152,13 +158,19 @@ def test_create_device_oled():
         display = display_name
 
     module_mock = Mock()
+    module_mock.oled.device.oled1234.return_value = display_name
     with patch.dict('sys.modules', **{
+            # mock luma.oled package
             'luma': module_mock,
             'luma.oled': module_mock,
             'luma.oled.device': module_mock
         }):
-        with pytest.raises(error.UnsupportedPlatform):
-            cmdline.create_device(args, display_types=display_types)
+        try:
+            device = cmdline.create_device(args, display_types=display_types)
+            assert device == display_name
+        except error.UnsupportedPlatform as e:
+            # non-rpi platform
+            pytest.skip(str(e))
 
 
 def test_create_device_lcd():
@@ -176,6 +188,7 @@ def test_create_device_lcd():
     module_mock = Mock()
     module_mock.lcd.device.lcd1234.return_value = display_name
     with patch.dict('sys.modules', **{
+            # mock spidev and luma.lcd packages
             'fake_gpio': module_mock,
             'spidev': module_mock,
             'luma': module_mock,
@@ -200,6 +213,7 @@ def test_create_device_led_matrix():
     module_mock = Mock()
     module_mock.led_matrix.device.matrix1234.return_value = display_name
     with patch.dict('sys.modules', **{
+            # mock spidev and luma.led_matrix packages
             'spidev': module_mock,
             'luma': module_mock,
             'luma.led_matrix': module_mock,
@@ -222,6 +236,7 @@ def test_create_device_emulator():
     module_mock = Mock()
     module_mock.emulator.device.emulator1234.return_value = display_name
     with patch.dict('sys.modules', **{
+            # mock spidev and luma.emulator packages
             'spidev': module_mock,
             'luma': module_mock,
             'luma.emulator': module_mock,

--- a/tests/test_cmdline.py
+++ b/tests/test_cmdline.py
@@ -115,7 +115,7 @@ def test_make_serial_spi():
         assert 'luma.core.interface.serial.spi' in repr(factory.spi())
     except error.UnsupportedPlatform as e:
         # non-rpi platform, e.g. ubuntu 64-bit
-        pytest.skip(str(e))
+        pytest.skip('{0} ({1})'.format(type(e).__name__, str(e)))
 
 
 def test_make_serial_spi_alt_gpio():
@@ -134,7 +134,7 @@ def test_make_serial_spi_alt_gpio():
             assert 'luma.core.interface.serial.spi' in repr(factory.spi())
         except error.DeviceNotFoundError as e:
             # non-rpi platform, e.g. ubuntu 64-bit
-            pytest.skip(str(e))
+            pytest.skip('{0} ({1})'.format(type(e).__name__, str(e)))
 
 
 def test_create_device():
@@ -170,7 +170,7 @@ def test_create_device_oled():
             assert device == display_name
         except error.UnsupportedPlatform as e:
             # non-rpi platform
-            pytest.skip(str(e))
+            pytest.skip('{0} ({1})'.format(type(e).__name__, str(e)))
 
 
 def test_create_device_lcd():

